pkg/cmd/roachtest/operations/changefeeds/pause.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package changefeeds
+
+import (
+	"context"
+	gosql "database/sql"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operations/helpers"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestflags"
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+// runPauseChangefeeds pauses a percentage of running changefeed jobs.
+func runPauseChangefeeds(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) registry.OperationCleanup {
+
+	// Connect to the cluster and ensure the connection is closed after use.
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer func() {
+		_ = conn.Close()
+	}()
+
+	// Fetch all running changefeed jobs.
+	allRunningCFJobs, err := fetchAllChangefeedJobsWithStatuses(ctx, conn, jobs.StatusRunning)
+	if err != nil {
+		o.Fatal(err)
+	}
+
+	// Retrieve the minimum number of changefeeds to keep running.
+	mcf, err := helpers.EnvOrDefaultInt(minChangefeeds, defaultEnvValuesInt[minChangefeeds])
+	if err != nil {
+		o.Fatal(err)
+	}
+
+	// If the number of changefeed jobs exceeds the minimum, start pausing jobs.
+	if len(allRunningCFJobs) >= mcf {
+		err = pauseChangefeeds(ctx, o, conn, allRunningCFJobs)
+		if err != nil {
+			o.Fatal(err)
+		}
+	} else {
+		o.Status(fmt.Sprintf("total running changefeeds %d are less than the minimum changefeeds criteria %d",
+			len(allRunningCFJobs), mcf))
+	}
+
+	// Return a nil cleanup function as there's no specific cleanup required here.
+	return nil
+}
+
+// pauseChangefeeds pauses a percentage of changefeed jobs.
+func pauseChangefeeds(
+	ctx context.Context, o operation.Operation, conn *gosql.DB, allCFJobs []*jobDetails,
+) error {
+
+	// Retrieve the percentage of changefeeds to pause from the environment.
+	pct, err := helpers.EnvOrDefaultInt(pctChangefeedsToPause, defaultEnvValuesInt[pctChangefeedsToPause])
```

Comment:
The pctChangefeedsToPause makes the code more complicated than it needs to be. In this operation we should just pause 1 changefeed that is able to pause.